src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketTagClientImpl.java

```diff
@@ -62,23 +62,20 @@ public Stream<BitbucketTag> getRemoteTags() {
     }
 
     public BitbucketTag getRemoteTag(String tagName) {
-        HttpUrl.Builder urlBuilder = null;
-        try {
-            urlBuilder = bitbucketRequestExecutor.getCoreRestPath().newBuilder()
+        HttpUrl.Builder urlBuilder = bitbucketRequestExecutor.getCoreRestPath().newBuilder()
                     .addPathSegment("projects")
                     .addPathSegment(projectKey)
                     .addPathSegment("repos")
                     .addPathSegment(repositorySlug)
-                    .addPathSegment("tags")
-                    .addPathSegment(URLEncoder.encode(tagName, "UTF-8"));
-        } catch (UnsupportedEncodingException e) {
-            throw new RuntimeException(e);
-        }
+                    .addPathSegment("commits")
+                    .addQueryParameter("until", tagName)
+                    .addQueryParameter("start", "0")
+                    .addQueryParameter("limit", "1");
 
-        HttpUrl url = urlBuilder.build();
-        BitbucketTag tag = bitbucketRequestExecutor.makeGetRequest(url, BitbucketTag.class).getBody();
 
-        return tag;
+        HttpUrl url = urlBuilder.build();
+        BitbucketCommit commits =  bitbucketRequestExecutor.makeGetRequest(url, BitbucketCommit.class).getBody();
```

Comment:
We discussed this offline, but the endpoint is returning a page of BitbucketCommits, which we can represent as `BitbucketPage<BitbucketCommit>`- so instead of having to extract the `values` array as part of the commit model, we can send a `TypeReference` to our request executor for the full page, and then fetch the values from that. Given we have those jackon representations, we may as well use them and I think that makes this a bit more extensible in the future (in case we ever do need multiple commits)